spring-cloud-stream-reactive/src/test/java/org/springframework/cloud/stream/reactive/StreamListenerReactiveTestInputOutputArgsWithSender.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.reactive;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import reactor.core.publisher.Flux;
+import rx.Observable;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.Input;
+import org.springframework.cloud.stream.annotation.Output;
+import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.support.MessageBuilder;
+
+/**
+ * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
+ */
+@RunWith(Parameterized.class)
+public class StreamListenerReactiveTestInputOutputArgsWithSender {
+
+	private Class<?> configClass;
+
+	public StreamListenerReactiveTestInputOutputArgsWithSender(Class<?> configClass) {
+		this.configClass = configClass;
+	}
+
+	@Parameterized.Parameters
+	public static Collection InputConfigs() {
+		return Arrays.asList(new Class[] {ReactorTestInputOutputArgsWithFluxSender1.class, RxJava1TestInputOutputArgsWithObservableSender1.class});
+	}
+
+	@Test
+	public void testInputOutputArgsWithFluxSender() throws Exception {
+		ConfigurableApplicationContext context = SpringApplication.run(this.configClass,
+				"--server.port=0");
+		// send multiple message
+		sendMessageAndValidate(context);
+		sendMessageAndValidate(context);
+		sendMessageAndValidate(context);
+		context.close();
+	}
+
+	@Test
+	public void testIncorrectUsage() {
```

Comment:
yeah, this was the result of the initial tests added when all possible `@StreamListener` method parameter/method annotations support was added and all the tests were inside a single test class. Then, I had parameterisation for all the test methods. After refactoring  and removing some of the supported options, the parameterisation diverged. will fix this.